provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/amqp/ContractAPIAmqp.java

```diff
@@ -134,6 +137,30 @@ public synchronized void requestNonceForTPM(RequestData requestData, ResponseCal
         this.provisioningAmqpOperations.open(registrationId, sslContext, this.amqpSaslHandler, this.useWebSockets);
     }
 
+    public synchronized void openWithSymmetricKeyAuthentication(RequestData requestData, ResponseCallback responseCallback, Object authorizationCallbackContext) throws ProvisioningDeviceClientException
+    {
+        if (requestData == null)
+        {
```

Comment:
This breaks he design. Why do you need this separate open?

Reply:
You should be able to identify the type of flow from security provider type.

Reply:
I'm not sure I need this new api anymore. I'll get rid of it

Reply:
Updated the code to no longer need that new API. The interface gets a bit clunky though since AMQP and MQTT both open and authenticate at the same time for symmetric key, but AMQP opens seperately from authentication during TPM flow. We may need to revisit this interface later